src/current/v24.3/ldap.md

```diff
@@ -0,0 +1,206 @@
+~~~
+
+#### Step 4: Create Database Users
+
+Q: Confirm: Does a new customer need to create one user here or all users in bulk? 
+
+Create a CockroachDB user that matches the LDAP username:
+
+~~~ sql
+CREATE ROLE username LOGIN;
+~~~
+
+### Connecting with LDAP Authentication
+
+Users can connect using their LDAP credentials:
+
+~~~ bash
+# Method 1: Password in connection string
+cockroach sql --url "postgresql://username:ldap_password@host:26257" --certs-dir=certs
+
+# Method 2: Password in environment variable
+export PGPASSWORD='ldap_password'
+cockroach sql --url "postgresql://username@host:26257" --certs-dir=certs
+~~~
+
+## LDAP Authorization
+
+### Overview
+
+LDAP authorization allows CockroachDB to use LDAP/AD groups as the source of truth for [role](https://www.cockroachlabs.com/docs/stable/security-reference/authorization#roles) assignments. When enabled:
+
+1. During login, CockroachDB fetches the user's group memberships from LDAP.
+2. Groups are mapped to CockroachDB roles using the group's Common Name (CN).
+3. The user is automatically granted membership to corresponding roles.
+4. Any existing role memberships not matched to LDAP groups are revoked.
+
+### Configuration
+
+#### Step 1: Enable LDAP Authorization
+
+Add the `ldapgrouplistfilter` parameter to your HBA configuration:
+
+~~~ sql
+SET CLUSTER SETTING server.host_based_authentication.configuration = '
+host    all    all    all    ldap    ldapserver=ldap.example.com 
+    ldapport=636 
+    "ldapbasedn=ou=users,dc=example,dc=com" 
+    "ldapbinddn=cn=readonly,dc=example,dc=com" 
+    ldapbindpasswd=readonly_password 
+    ldapsearchattribute=uid 
+    "ldapsearchfilter=(memberof=cn=cockroachdb_users,ou=groups,dc=example,dc=com)"
+    "ldapgrouplistfilter=(objectClass=groupOfNames)"
+host    all    root    0.0.0.0/0    password';
+~~~
+
+The `ldapgrouplistfilter` parameter defines which LDAP objects should be considered as groups.
+
+Q: Get more specific/realistic example from Eng?
+
+#### Step 2: Create Matching Roles
+
+Create CockroachDB roles that match your LDAP group names and grant appropriate privileges to each role.
+
+For an LDAP group `cn=db_admins,ou=groups,dc=example,dc=com`:
+
+~~~ sql
+CREATE ROLE db_admins;
+GRANT ALL ON DATABASE app TO db_admins;
+~~~
+
+### Group Search Template
+
+You can configure how LDAP groups are searched using the `server.ldap.group_search_template` setting:
+
+~~~ sql
+SET CLUSTER SETTING server.ldap.group_search_template = 'ou=groups,dc=example,dc=com??sub?(&(objectClass=groupOfNames)(member={USER_DN}))';
+~~~
+
+The template follows RFC4515/4516 format:
+
+- Base DN: Where to start the search
+- Scope: How deep to search (`sub` for subtree)
+- Filter: Criteria for matching groups
+- `{USER_DN}`: Placeholder for the authenticated user's DN
+
+### Troubleshooting
+
+Enable authentication logging to troubleshoot LDAP issues:
+
+~~~ sql
+SET CLUSTER SETTING server.auth_log.sql_sessions.enabled = true;
```

Comment:
We should have an example of how to get the log from `cockroach-data` folder and one sample log to help in debugging. `tail -f cockroach-session.log`. Would be okay to point to a different wiki as well.

Reply:
This already exists but needs to be enabled. Mentioned here: https://www.cockroachlabs.com/docs/v24.2/logging#sessions

I found customers were confused how to enable it and later use it to get logs for a specific user via grep. Since error logs are redacted, the actual errors might not be visible to log monitoring dashboards and would need to be debugged on the node itself.

Reply:
It's an existing log file. But a particular setting has to be enabled before we write additional lines to that file